### PR TITLE
pkg/operator: Add unit tests

### PR DESCRIPTION
**Pull request by sgreene570**
**- What I did**
Added a couple more unit tests to `pkg/operator`
**- How to verify it**
`go test -v github.com/openshift/machine-config-operator/pkg/operator`
**- Description for the changelog**

pkg/operator add unit tests

### CODE REVIEW

File: pkg/operator/render_test.go [hunk 1]
```diff
@@ -35,3 +35,41 @@ func TestClusterDNSIP(t *testing.T) {
...
+					t.Fatalf("%s failed: %s", desc, err.Error())
+				} else {
+					return
+				}
+			}
+			if template == nil || len(template) == 0 {
+				t.Fatalf("Template is empty")
+			}
+		})
+	}
```
[the remark this answers is not in the document]
Comment (kikisdeliveryservice) [reply pkg/operator/render_test.go]: what if the func returns an unexpected byte[]

[the remark this answers is not in the document]
Comment (sgreene570) [reply pkg/operator/render_test.go]: Added a check to make sure that byte array can be safely turned back into a string (this should be a good way to check for strange byte array data).
Also, if the template is rendered with a field missing, `renderAsset` will return with err.

File: pkg/operator/status_test.go [hunk 2]
```diff
@@ -531,3 +531,32 @@ func TestInClusterBringUpStayOnErr(t *testing.T) {
 
 	assert.False(t, optr.inClusterBringup)
 }
+
+func TestInitalizeOperator(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sgreene570) [reply pkg/operator/status_test.go]: removed this test from PR

File: pkg/operator/render_test.go [hunk 3]
```diff
@@ -35,3 +35,60 @@ func TestClusterDNSIP(t *testing.T) {
...
+		RenderConfig: &renderConfig{
+			TargetNamespace: "testing-namespace",
+		},
+	}, {
+		// Nested field test
+		Path: "manifests/machineconfigcontroller/deployment.yaml",
+		RenderConfig: &renderConfig{
+			TargetNamespace: "testing-namespace",
+			Images: &Images{
+				MachineConfigController: "{MCC: PLACEHOLDER}"},
```
Comment (cgwalters) [inline pkg/operator/render_test.go]: Looks reasonable to me but I think we could improve this by adding a member to the test struct like in this case `FindExpected: "  namespace: testing-namespace"` or so and have the test code assert that string exists in the output if `FindExpected != nil`?

Comment (sgreene570) [reply pkg/operator/render_test.go]: Sure, can do.

**Comment by ashcrow** [comment]
Thank you for this PR!

**Pull request merged**